src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/CpsPackageReferenceProject.cs

```diff
@@ -68,9 +68,6 @@ public class CpsPackageReferenceProject : PackageReferenceProject<List<Framework
             InternalMetadata.Add(NuGetProjectMetadataKeys.UniqueName, ProjectUniqueName);
             InternalMetadata.Add(NuGetProjectMetadataKeys.FullPath, ProjectFullPath);
             InternalMetadata.Add(NuGetProjectMetadataKeys.ProjectId, projectId);
-
-            InstalledPackages = new List<FrameworkInstalledPackages>();
```

Comment:
Ex. why was this removed. It's not super obvious.

Reply:
One reason was there was shared code which calls below and multithreaded env causing [problems](https://user-images.githubusercontent.com/49205731/156683251-76eaba45-dfad-4108-a857-4a7d8748c339.png).

https://github.com/NuGet/NuGet.Client/blob/9b7714d5e8f9509488158e335b069472d08aa937/src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/PackageReferenceProject.cs#L119-L120

Instead of using instance initialized at constructor we can create new temp instance per request then populate it.

https://github.com/NuGet/NuGet.Client/pull/4498/files#diff-cf69ebf676e61580349108d38e5dab1bbff91468194c6a8d342149d264139b15R124-R125

Once populated we make it cache.
https://github.com/NuGet/NuGet.Client/pull/4498/files#diff-cf69ebf676e61580349108d38e5dab1bbff91468194c6a8d342149d264139b15R197-R201